src/libraries/System.Text.Json/tests/JsonNode/JsonArrayTests.cs

```diff
@@ -0,0 +1,432 @@
+
+            Assert.False(jArray.Remove(node2)); // remove an already removed node.
+            Assert.Equal(2, jArray.Count);
+
+            // Contains
+            Assert.True(jArray.Contains(node1));
+            Assert.False(jArray.Contains(node2));
+            Assert.True(jArray.Contains(node3));
+
+            // Insert
+            jArray.Insert(1, node2);
+            Assert.Equal(3, jArray.Count);
+
+            Assert.Equal(0, jArray.IndexOf(node1));
+            Assert.Equal(1, jArray.IndexOf(node2));
+            Assert.Equal(2, jArray.IndexOf(node3));
+        }
+
+        [Fact]
+        public static void CopyTo()
+        {
+            JsonNode node1 = 1;
+            JsonNode node2 = 2;
+
+            IList<JsonNode?> jArray = new JsonArray(node1, node2, null);
+            var arr = new JsonNode[4];
+            jArray.CopyTo(arr, 0);
+
+            Assert.Same(node1, arr[0]);
+            Assert.Same(node2, arr[1]);
+            Assert.Null(arr[2]);
+
+            arr = new JsonNode[5];
+            jArray.CopyTo(arr, 1);
+            Assert.Null(arr[0]);
+            Assert.Same(node1, arr[1]);
+            Assert.Same(node2, arr[2]);
+            Assert.Null(arr[3]);
+            Assert.Null(arr[4]);
+
+            arr = new JsonNode[3];
+            Assert.Throws<ArgumentException>(() => jArray.CopyTo(arr, 1));
+        }
+
+        [Fact]
+        public static void ReAddSameNode_Throws()
+        {
+            var jValue = JsonValue.Create(1);
+
+            var jArray = new JsonArray(jValue);
+            Assert.Throws<InvalidOperationException>(() => jArray.Add(jValue));
+        }
+
+        [Fact]
+        public static void ReAddRemovedNode()
+        {
+            var jValue = JsonValue.Create(1);
+
+            var jArray = new JsonArray(jValue);
+            Assert.Equal(1, jArray.Count);
+            jArray.Remove(jValue);
+            Assert.Equal(0, jArray.Count);
+            jArray.Add(jValue);
+            Assert.Equal(1, jArray.Count);
+        }
+
+        [Fact]
+        public static void CreatingJsonArrayFromNodeArray()
+        {
+            JsonNode[] expected = { "sushi", "pasta", "cucumber soup" };
+
+            var dishesJsonArray = new JsonArray(expected);
+            Assert.Equal(3, dishesJsonArray.Count);
+
+            for (int i = 0; i < dishesJsonArray.Count; i++)
+            {
+                Assert.Equal(expected[i], dishesJsonArray[i]);
+            }
+        }
+
+        [Fact]
+        public static void CreatingJsonArrayFromArrayOfStrings()
+        {
+            var strangeWords = new string[]
+            {
+                "supercalifragilisticexpialidocious",
+                "gladiolus",
+                "albumen",
+                "smaragdine",
+            };
+
+            var strangeWordsJsonArray = new JsonArray();
+            strangeWords.Where(word => word.Length < 10).
+                ToList().ForEach(str => strangeWordsJsonArray.Add(JsonValue.Create(str)));
```

Comment:
Yes since ForEach exists on `List<>`, not `IEnumerable<T>`.

We could add additional constructors to JsonArray to take an `IEnumerable<T>` or `IEnumerable<string>` etc to make this easier and not require ForEach.

Reply:
Yup we can wait for user feedback here.